examples/kitchen-sink.yaml

```
@@ -416,7 +419,23 @@ resource:
     # Configure `service.name` resource attribute
     #
     # Environment variable: OTEL_SERVICE_NAME
-    service.name: !!str "unknown_service"
+    - name: service-name
+      string_value: "unknown_service"
+    # Configure other resource attributes with explicit types.
+    - name: bool_key
+      bool_value: true
+    - name: int_key
+      int_value: 1
```

I kind of the modeling below since it simplifies the common case:
```suggestion
    - name: service-name
      value: "unknown_service"
    # Configure other resource attributes with explicit types.
    - name: bool_key
      value: true
      type: bool
    - name: int_key
      value: 1
      type: int
```

---

That's a nice way to model it 👍

---

I prefer (name, value, type) as well.

Schema wise, not sure if it will be possible to represent a bool value as just `true`, or if it needs to always be a string as `"true"`, with the config.yaml implementation having to convert a string to the proper type, but even then, a triplet (name, value, type) is better.

To address with usage examples.

---

> Schema wise, not sure if it will be possible to represent a bool value as just true, or if it needs to always be a string as "true", with the config.yaml implementation having to convert a string to the proper type, but even then, a triplet (name, value, type) is better.

And what about arrays of strings, arrays of bools, arrays of ints, arrays of doubles? Same question applies. We could use something like [oneOf](https://json-schema.org/understanding-json-schema/reference/combining#oneOf) and list the allowed types, or treat all values as a string and require arrays to be comma separated and require implementations to parse the string value to the specified `type`.

---

FYI, I pushed a commit which sets the `.resource.attributes[]` items schema to be:
```
{
  "title": "AttributeNameValue",
  "type": "object",
  "additionalProperties": false,
  "properties": {
      "name": {
          "type": "string"
      },
      "value": {
          "oneOf": [
              {"type": "string"},
              {"type": "number"},
              {"type": "boolean"},
              {"type": "null"},
              {"type": "array", "items": {"type": "string"}},
              {"type": "array", "items": {"type": "boolean"}},
              {"type": "array", "items": {"type": "number"}}
          ]
      },
      "type": {
          "enum": [
              null,
              "string",
              "bool",
              "int",
              "double",
              "string_array",
              "bool_array",
              "int_array",
              "double_array"
          ]
      }
  },
  "required": [
      "name", "value"
  ]
}
```
Notes:
- `.type` can be null, or explicitly set to one of the enumerated values corresponding to the allowed types for resource attributes. 
- If `.type` null, it defaults to be `string`. This behavior is conveyed to users / implementers via a comment in the `kitchen-sink.yaml` example. Note, with complicated semantics like this, #104 is important since it helps ensure consistent comments across all the example files.
- `.value` must be "oneOf" the allowed types. Its up to the implementation to interpret it as the specified `.type`. For example, in the java implementation this "oneOf" results in `.value` being a generic `Object`. We'll need to cast `Object` to the corresponding `.type`, and fail if it can't be cast. 

This all feels a little more brittle and difficult to implement than the first commit in this PR, where we have separate properties for each of the types (i.e. `bool_value`, `string_value`, `int_value`, `double_value`, etc). But it does feel a little better for the simple case.

---

I agree the schema is more complex.

Some general observation:

* the yaml schema is written once, by maintainers of this repository.
* the yaml schema will be implemented a dozen times, in each language SDK.
* the yaml config files will be written many, many times, by many people. At scale, it means for every deployment of opentelemetry.

Each time we make the config.yaml more natural for end users, it is a win that will increase adoption and contribute to the overall success of this project, even at the cost of some pain in the schema or for language maintainers (who have to support the schema).

An example of this is #110, with more complexity in the schema, and less chances for users to have invalid configs.

---

@marcalff I agree and think that the current state of this PR is the best path forward.